### PR TITLE
Added rule: "Do you check your API serialisation format?"

### PR DESCRIPTION
**Pull request by christoment**
>
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

Things I found during migrating TimePro to .NET 8.

> 2. What was changed?

Added rule - Do you check your API serialisation format?

> 3. Did you do pair or mob programming (list names)?
No

### CODE REVIEW

File: rules/do-you-check-your-api-serialisation-format/rule.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+The primary reason for switching to `System.Text.Json` is its [faster performance and lower memory usage](https://devblogs.microsoft.com/dotnet/the-convenience-of-system-text-json/) compared to `Newtonsoft.Json`. However, it also breaks compatibility and lacks some features found in `Newtonsoft.Json`.
+
+
+## The differences
+[This Microsoft documentation](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft?pivots=dotnet-9-0#table-of-differences) contains the compiled list of the difference between `System.Text.Json` and `Newtonsoft.Json`.
+
+
+### Notable Things to check
+
+- ⚠️ **Default serialisation proerty name casing**.
```
Comment (william-liebenberg) [inline rules/do-you-check-your-api-serialisation-format/rule.md:37]: Typo: should be 'property'

**Review by william-liebenberg — CHANGES_REQUESTED** [review]

File: rules/do-you-check-your-api-serialisation-format/rule.md (hunk 1 above)
Comment (christoment) [reply rules/do-you-check-your-api-serialisation-format/rule.md]: Fixed :)

**Review by william-liebenberg — APPROVED** [review]
👍

**Review by jernejk — APPROVED** [review]
LGTM.

The .NET Framework story is a bit less trivial because you have Katana, Owin, ASP.NET, ASP.NET Core and probably more. I remember some combination gave you camelCase by default.

I think, it's good as it is in order to not complicate it any further. 😄 (and that it slowly drifts into archive)

**Pull request merged**